bundles/org.openhab.binding.bluetooth.bluegiga/src/main/java/org/openhab/binding/bluetooth/bluegiga/handler/BlueGigaBridgeHandler.java

```diff
@@ -249,16 +238,70 @@ public void initialize() {
     @Override
     public void dispose() {
         try {
+            stopScheduledTasks();
             BlueGigaSerialHandler bgh = getBgHandler();
             bgh.removeEventListener(this);
             bgh.removeHandlerListener(this);
             bgh.close();
+            initComplite = false;
         } catch (IllegalStateException e) {
             // ignore if handler wasn't set at all
         }
         closeSerialPort();
     }
 
+    private void startScheduledTasks() {
+        passiveScanIdleTimer.schedule(configuration.passiveScanIdleTime);
+        logger.debug("Start scheduled task to remove old devices ones per every hour");
+        removeOldDevicesTask = scheduler.scheduleWithFixedDelay(this::removeOldDevices, 1, 1, TimeUnit.HOURS);
+    }
+
+    private void stopScheduledTasks() {
+        passiveScanIdleTimer.cancel();
+        passiveScanIdleTimer = null;
+        removeOldDevicesTask.cancel(true);
+    }
+
+    private void removeOldDevices() {
+        logger.debug("Check old devices, count {}", devices.size());
+        devices.forEach((address, device) -> {
+            DateTime lastCommunicationTime = ((BlueGigaBluetoothDevice) device).getLastCommunicationTime();
+            logger.debug(" Device {} last communication time {}", address, lastCommunicationTime);
+            if (lastCommunicationTime.plusDays(1).isBeforeNow()) {
```

Comment:
A day is a long time for devices to hang around, especially when some of them generate new mac addresses every two minutes. That means you would track and keep memory for over 720+ devices every day. I'm using 5 minutes in Bluez, is there a reason you are tracking it for longer?
Also, you haven't taken into account devices which may have handlers and thus hold a reference to a bluetooth device. If the BlueGigaBridgeHandler loses its reference to that bluetooth device reference then that handler is holding onto a dead device instance that will never receive events until Openhab restarts or the handler is disabled then enabled again.
I solved this problem by adding a `hasListeners` method to the `BluetoothDevice` class. It is currently in my PR but I think you would likely need it as well so we need to coordinate on that.

Reply:
Well, day is not much compared to current implementation where cleanup is not done at all ;) There could also be devices which use static mac, but isn't available all the time, so 5 min cleanup time could be rather aggressive as well.